### PR TITLE
feat: React - NavBar and Login UI (#40950)

### PR DESCRIPTION
**Pull request by Ankush-Negi**

### CODE REVIEW

File: src/pages/Login/Login.jsx [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+import CssBaseline from '@material-ui/core/CssBaseline';
+import TextField from '@material-ui/core/TextField';
+import LockOutlinedIcon from '@material-ui/icons/LockOutlined';
+import Typography from '@material-ui/core/Typography';
+import { withStyles } from '@material-ui/styles';
+import InputAdornment from '@material-ui/core/InputAdornment';
+import Grid from '@material-ui/core/Grid';
+import EmailIcon from '@material-ui/icons/Email';
+import PropTypes from 'prop-types';
+import VisibilityOffIcon from '@material-ui/icons/VisibilityOff';
```
Comment (VisaaliSuccessive) [inline src/pages/Login/Login.jsx:13]: Irregular import modules and component

File: src/pages/Trainee/components/AddDialog/AddDialog.jsx [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+    this.setState({ email: e.target.value });
+  };
+
+  handlePasswordChange = (e) => {
+    this.setState({ password: e.target.value });
+  };
+
+  handleConfirmPasswordChange = (e) => {
+    this.setState({ confirmPassword: e.target.value });
+  };
```
Comment (VisaaliSuccessive) [inline src/pages/Trainee/components/AddDialog/AddDialog.jsx]: Can use single method or handler for updating values

File: src/pages/Trainee/components/AddDialog/helper.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+const ValidationSchema = yup.object().shape({
+  Name: yup
+    .string()
+    .required('Name is a required field')
+    .min(3),
+  Email: yup
+    .string()
+    .required('Email Address is a required field')
+    .matches('^[a-z0-9](.?[a-z0-9]){5,}@g(oogle)?mail.com$', { message: 'Email address must be a valid email' }),
```
Comment (VisaaliSuccessive) [inline src/pages/Trainee/components/AddDialog/helper.js:11]: can use yup email validation

File: src/pages/Trainee/components/AddDialog/helper.js [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+  Password: yup
+    .string()
+    .required('Password is required')
+    .min(8, 'Must contain 8 characters, atleast one uppercase letter, one lowercase letter and a number'),
```
Comment (VisaaliSuccessive) [inline src/pages/Trainee/components/AddDialog/helper.js]: Wrong validation of password

File: src/pages/Trainee/components/AddDialog/AddDialog.jsx [hunk 5]
```diff
@@ -0,0 +1,223 @@
...
+      return allErrors[field];
+    }
+    return false;
+  }
+
+  isTouched = (value) => {
+    const { touch } = this.state;
+    delete touch[value];
+    this.setState({ touch });
+  };
```
Comment (VisaaliSuccessive) [inline src/pages/Trainee/components/AddDialog/AddDialog.jsx:93]: current approach you have used is a bit redundant so make it in a better way.

File: src/pages/Login/helper.js [hunk 6]
```diff
@@ -0,0 +1,13 @@
+import * as yup from 'yup';
+
+const LoginValidationSchema = yup.object().shape({
+  Email: yup
+    .string()
+    .required('Email Address is a required field')
+    .matches('^[a-z0-9](.?[a-z0-9]){5,}@g(oogle)?mail.com$', { message: 'Email address must be a valid email' }),
```
Comment (VisaaliSuccessive) [inline src/pages/Login/helper.js:7]: can use yup email validation

File: src/pages/Login/Login.jsx [hunk 7]
```diff
@@ -0,0 +1,188 @@
...
+      return allErrors[field];
+    }
+    return false;
+  }
+
+  isTouched = (value) => {
+    const { touch } = this.state;
+    delete touch[value];
+    this.setState({ touch });
+  };
```
Comment (VisaaliSuccessive) [inline src/pages/Login/Login.jsx:114]: current approach you have used is making too much repeated call, so make it in a better way

**Review by VisaaliSuccessive — COMMENTED** [review]
Fix lint errors.
Use es6 or beyond approach in components like Button.
Improve validation logic in file 'InputDemo.jsx'.

**Comment by VisaaliSuccessive** [comment]
On the whole, need to use the features of es6 and beyond instead of using the older features of JS.

**Comment by Ankush-Negi** [comment]
Contains all file in newest PR.
confirmed with @visaali

**Pull request closed without merging**